Option for host and port on `make docs-live`

i tried run `make docs-live` but it tried to use `0.0.0.0:8080` which i use for other thing.

is there option to set sphinx-autobuild port and host?

or is there any other way to set that with current makefile?

<details>
  <summary>error log</summary>
  
```python
$ export SEARX_DEBUG=1 && make docs-live
build searx/brand.py
build utils/brand.env
PYENV     usage: source ./local/py3/bin/activate
PYENV     using virtualenv from ./local/py3
PYENV     install .
SPHINX    docs-live --> file:///home/r3r/git/searx/dist/docs
+--------------------------------- manual build ---------------------------------
| Running Sphinx v3.2.1
DEBUG:searx:read configuration from /home/r3r/git/searx/searx/settings.yml
INFO:searx:Initialisation done
INFO:searx.plugins.https_rewrite:29 rules loaded
INFO:searx.search:max_request_timeout=None
DEBUG:searx.webapp:static directory is /home/r3r/git/searx/searx/static
DEBUG:searx.webapp:templates directory is /home/r3r/git/searx/searx/templates
DEBUG:searx.engines:wikidata engine: Starting background initialization
DEBUG:searx.engines:gigablast engine: Starting background initialization
DEBUG:searx.engines:soundcloud engine: Starting background initialization
DEBUG:searx.engines:yggtorrent engine: Starting background initialization
DEBUG:searx.engines:wolframalpha engine: Starting background initialization
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): query.wikidata.org:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): soundcloud.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): gigablast.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www2.yggtorrent.si:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.wolframalpha.com:443
| loading pickled environment... done
| building [mo]: targets for 0 po files that are out of date
| building [html]: targets for 0 source files that are out of date
| updating environment: [config changed ('jinja_contexts')] 41 added, 0 changed, 0 removed
| reading sources... [  2%] admin/api
| reading sources... [  4%] admin/architecture
| reading sources... [  7%] admin/buildhosts
| reading sources... [  9%] admin/engines
| reading sources... [ 12%] admin/filtron
| reading sources... [ 14%] admin/index
| reading sources... [ 17%] admin/installation
| reading sources... [ 19%] admin/installation-apache
DEBUG:urllib3.connectionpool:https://query.wikidata.org:443 "GET /sparql?query=%0ASELECT+%3Fitem+%3Fname%0AWHERE+%7B%0A++++%7B%0A++++++SELECT+%3Fitem%0A++++++WHERE+%7B+%3Fitem+wdt%3AP279%2A+wd%3AQ12132+%7D%0A++++%7D+UNION+%7B%0A++++++VALUES+%3Fitem+%7B+wd%3AP571+wd%3AP576+wd%3AP580+wd%3AP582+wd%3AP569+wd%3AP570+wd%3AP619+wd%3AP620+wd%3AP27+wd%3AP495+wd%3AP17+wd%3AP159+wd%3AP36+wd%3AP35+wd%3AP6+wd%3AP122+wd%3AP37+wd%3AP1082+wd%3AP2046+wd%3AP281+wd%3AP38+wd%3AP2048+wd%3AP400+wd%3AP50+wd%3AP170+wd%3AP57+wd%3AP175+wd%3AP178+wd%3AP162+wd%3AP176+wd%3AP58+wd%3AP272+wd%3AP264+wd%3AP123+wd%3AP449+wd%3AP750+wd%3AP86+wd%3AP577+wd%3AP136+wd%3AP364+wd%3AP212+wd%3AP957+wd%3AP275+wd%3AP277+wd%3AP348+wd%3AP840+wd%3AP1098+wd%3AP282+wd%3AP1018+wd%3AP218+wd%3AP169+wd%3AP112+wd%3AP1454+wd%3AP137+wd%3AP1029+wd%3AP225+wd%3AP274+wd%3AP1346+wd%3AP1120+wd%3AP498+wd%3AP856+wd%3AP1324+wd%3AP1581+%7D%0A++++%7D%0A++++OPTIONAL+%7B+%3Fitem+rdfs%3Alabel+%3Fname.+%7D%0A%7D%0A HTTP/1.1" 200 None
| reading sources... [ 21%] admin/installation-docker
| reading sources... [ 24%] admin/installation-nginx
| reading sources... [ 26%] admin/installation-searx
DEBUG:urllib3.connectionpool:https://soundcloud.com:443 "GET / HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): a-v2.sndcdn.com:443
| reading sources... [ 29%] admin/installation-uwsgi
| reading sources... [ 31%] admin/morty
| reading sources... [ 34%] admin/plugins
| reading sources... [ 36%] admin/settings
| reading sources... [ 39%] admin/update-searx
| reading sources... [ 41%] blog/admin
DEBUG:urllib3.connectionpool:https://www2.yggtorrent.si:443 "GET / HTTP/1.1" 307 None
DEBUG:searx.engines:yggtorrent engine: Initialized
DEBUG:urllib3.connectionpool:https://gigablast.com:443 "GET /search?c=main&qlangcountry=en-us&q=south&s=10 HTTP/1.1" 200 2709
DEBUG:searx.engines:gigablast engine: Initialized
| reading sources... [ 43%] blog/command-line-engines
| reading sources... [ 46%] blog/index
| reading sources... [ 48%] blog/intro-offline
| reading sources... [ 51%] blog/lxcdev-202006
| reading sources... [ 53%] blog/private-engines
| reading sources... [ 56%] blog/python3
| reading sources... [ 58%] dev/contribution_guide
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/13-6877fc25-3.js HTTP/1.1" 200 None
| reading sources... [ 60%] dev/engine_overview
| reading sources... [ 63%] dev/index
DEBUG:urllib3.connectionpool:https://www.wolframalpha.com:443 "GET /input/api/v1/code?ts=9999999999999999999 HTTP/1.1" 200 None
DEBUG:searx.engines:wolframalpha engine: Initialized
| reading sources... [ 65%] dev/makefile
| reading sources... [ 68%] dev/plugins
| reading sources... [ 70%] dev/quickstart
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/48-e0f5f1c7-3.js HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/0-6516b28c-3.js HTTP/1.1" 200 None
| reading sources... [ 73%] dev/reST
| reading sources... [ 75%] dev/search_api
| reading sources... [ 78%] dev/translation
| reading sources... [ 80%] index
| reading sources... [ 82%] user/index
| reading sources... [ 85%] user/own-instance
| reading sources... [ 87%] user/search_syntax
| reading sources... [ 90%] utils/filtron.sh
| reading sources... [ 92%] utils/index
| reading sources... [ 95%] utils/lxc.sh
| reading sources... [ 97%] utils/morty.sh
| reading sources... [100%] utils/searx.sh
|
WARNING: Unexpected return code 42 from command '../utils/lxc.sh --help'
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/2-46c8af5f-3.js HTTP/1.1" 200 None
DEBUG:searx.wikidata:request time 0:00:00.846739
DEBUG:searx.engines:wikidata engine: Initialized
| looking for now-outdated files... none found
| pickling environment... done
| checking consistency... done
| preparing documents... done
| writing output... [  2%] admin/api
| writing output... [  4%] admin/architecture
| writing output... [  7%] admin/buildhosts
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/1-29bdd2fe-3.js HTTP/1.1" 200 None
| writing output... [  9%] admin/engines
| writing output... [ 12%] admin/filtron
| writing output... [ 14%] admin/index
| writing output... [ 17%] admin/installation
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/47-ae1d61aa-3.js HTTP/1.1" 200 None
| writing output... [ 19%] admin/installation-apache
| writing output... [ 21%] admin/installation-docker
| writing output... [ 24%] admin/installation-nginx
| writing output... [ 26%] admin/installation-searx
| writing output... [ 29%] admin/installation-uwsgi
| writing output... [ 31%] admin/morty
| writing output... [ 34%] admin/plugins
| writing output... [ 36%] admin/settings
| writing output... [ 39%] admin/update-searx
| writing output... [ 41%] blog/admin
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/46-72510647-3.js HTTP/1.1" 200 None
| writing output... [ 43%] blog/command-line-engines
| writing output... [ 46%] blog/index
| writing output... [ 48%] blog/intro-offline
| writing output... [ 51%] blog/lxcdev-202006
| writing output... [ 53%] blog/private-engines
| writing output... [ 56%] blog/python3
| writing output... [ 58%] dev/contribution_guide
| writing output... [ 60%] dev/engine_overview
| writing output... [ 63%] dev/index
DEBUG:searx.engines:soundcloud engine: Initialized
| writing output... [ 65%] dev/makefile
| writing output... [ 68%] dev/plugins
| writing output... [ 70%] dev/quickstart
| writing output... [ 73%] dev/reST
| writing output... [ 75%] dev/search_api
| writing output... [ 78%] dev/translation
| writing output... [ 80%] index
| writing output... [ 82%] user/index
| writing output... [ 85%] user/own-instance
| writing output... [ 87%] user/search_syntax
| writing output... [ 90%] utils/filtron.sh
| writing output... [ 92%] utils/index
| writing output... [ 95%] utils/lxc.sh
| writing output... [ 97%] utils/morty.sh
| writing output... [100%] utils/searx.sh
|
| generating indices...  genindexdone
| writing additional pages...  searchdone
| copying images... [ 20%] admin/arch_public.dot
| copying images... [ 40%] blog/searx-admin-engines.png
| copying images... [ 60%] blog/searxpy3.png
| copying images... [ 80%] dev/svg_image.svg
| copying images... [100%] dev/hello.dot
|
| copying downloadable files... [100%] dev/reST.rst
|
| copying static files... ... done
| copying extra files... done
| dumping search index in English (code: en)... done
| dumping object inventory... done
| build succeeded, 1 warning.
|
| The HTML pages are in dist/docs.
| Copying tabs assets
+--------------------------------------------------------------------------------
[I 201029 11:30:05 server:335] Serving on http://0.0.0.0:8080
Traceback (most recent call last):
  File "./local/py3/bin/sphinx-autobuild", line 8, in <module>
    sys.exit(main())
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/sphinx_autobuild/cli.py", line 188, in main
    server.serve(port=portn, host=args.host, root=outdir, open_url_delay=args.delay)
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/livereload/server.py", line 339, in serve
    self.application(
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/livereload/server.py", line 291, in application
    app.listen(port, address=host)
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/tornado/web.py", line 2116, in listen
    server.listen(port, address)
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/tornado/tcpserver.py", line 151, in listen
    sockets = bind_sockets(port, address=address)
  File "/home/r3r/git/searx/local/py3/lib/python3.8/site-packages/tornado/netutil.py", line 174, in bind_sockets
    sock.bind(sockaddr)
OSError: [Errno 98] Address already in use
make: *** [Makefile:88: docs-live] Error 1
```
</details>

e:

i try to run sphinx-autobuild directly but i got import error and it fail when i visit the site

<details>
  <summary>error log</summary>
  
```python
$ sphinx-autobuild docs _build
+--------------------------------- manual build ---------------------------------
| Running Sphinx v3.2.1

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/r3r/envs/searx/lib/python3.8/site-packages/sphinx/config.py", line 319, in eval_config_file
    execfile_(filename, namespace)
  File "/home/r3r/envs/searx/lib/python3.8/site-packages/sphinx/util/pycompat.py", line 89, in execfile_
    exec(code, _globals)
  File "/home/r3r/git/searx/docs/conf.py", line 4, in <module>
    from sphinx_build_tools import load_sphinx_config
ModuleNotFoundError: No module named 'sphinx_build_tools'

+--------------------------------------------------------------------------------
[I 201029 12:17:05 server:335] Serving on http://127.0.0.1:8000
[I 201029 12:17:05 handlers:62] Start watching changes
[I 201029 12:17:05 handlers:64] Start detecting changes
[W 201029 12:17:13 web:2246] 404 GET / (127.0.0.1) 0.74ms
[W 201029 12:17:14 web:2246] 404 GET /favicon.ico (127.0.0.1) 2.21ms
```
</details>

e2:

the only way to view docs right now is to run `make docs` after i edit doc file

<details>
  <summary>log</summary>
  
```python
$ make docs
build searx/brand.py
build utils/brand.env
PYENV     usage: source ./local/py3/bin/activate
PYENV     using virtualenv from ./local/py3
PYENV     install .
SPHINX    docs --> file:///home/r3r/git/searx/dist/docs
Running Sphinx v3.2.1
DEBUG:searx:read configuration from /home/r3r/git/searx/searx/settings.yml
INFO:searx:Initialisation done
INFO:searx.plugins.https_rewrite:29 rules loaded
INFO:searx.search:max_request_timeout=None
DEBUG:searx.webapp:static directory is /home/r3r/git/searx/searx/static
DEBUG:searx.webapp:templates directory is /home/r3r/git/searx/searx/templates
DEBUG:searx.engines:wikidata engine: Starting background initialization
DEBUG:searx.engines:gigablast engine: Starting background initialization
DEBUG:searx.engines:soundcloud engine: Starting background initialization
DEBUG:searx.engines:yggtorrent engine: Starting background initialization
DEBUG:searx.engines:wolframalpha engine: Starting background initialization
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): gigablast.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): soundcloud.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): query.wikidata.org:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.wolframalpha.com:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www2.yggtorrent.si:443
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: [config changed ('jinja_contexts')] 42 added, 0 changed, 0 removed
DEBUG:urllib3.connectionpool:https://soundcloud.com:443 "GET / HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): a-v2.sndcdn.com:443
DEBUG:urllib3.connectionpool:https://gigablast.com:443 "GET /search?c=main&qlangcountry=en-us&q=south&s=10 HTTP/1.1" 200 2709
DEBUG:urllib3.connectionpool:https://www2.yggtorrent.si:443 "GET / HTTP/1.1" 307 None
DEBUG:searx.engines:gigablast engine: Initialized
DEBUG:searx.engines:yggtorrent engine: Initialized
reading sources... [100%] utils/standalone_searx.py
WARNING: Unexpected return code 42 from command '../utils/lxc.sh --help'
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
DEBUG:urllib3.connectionpool:https://www.wolframalpha.com:443 "GET /input/api/v1/code?ts=9999999999999999999 HTTP/1.1" 200 None
DEBUG:searx.engines:wolframalpha engine: Initialized
DEBUG:urllib3.connectionpool:https://query.wikidata.org:443 "GET /sparql?query=%0ASELECT+%3Fitem+%3Fname%0AWHERE+%7B%0A++++%7B%0A++++++SELECT+%3Fitem%0A++++++WHERE+%7B+%3Fitem+wdt%3AP279%2A+wd%3AQ12132+%7D%0A++++%7D+UNION+%7B%0A++++++VALUES+%3Fitem+%7B+wd%3AP571+wd%3AP576+wd%3AP580+wd%3AP582+wd%3AP569+wd%3AP570+wd%3AP619+wd%3AP620+wd%3AP27+wd%3AP495+wd%3AP17+wd%3AP159+wd%3AP36+wd%3AP35+wd%3AP6+wd%3AP122+wd%3AP37+wd%3AP1082+wd%3AP2046+wd%3AP281+wd%3AP38+wd%3AP2048+wd%3AP400+wd%3AP50+wd%3AP170+wd%3AP57+wd%3AP175+wd%3AP178+wd%3AP162+wd%3AP176+wd%3AP58+wd%3AP272+wd%3AP264+wd%3AP123+wd%3AP449+wd%3AP750+wd%3AP86+wd%3AP577+wd%3AP136+wd%3AP364+wd%3AP212+wd%3AP957+wd%3AP275+wd%3AP277+wd%3AP348+wd%3AP840+wd%3AP1098+wd%3AP282+wd%3AP1018+wd%3AP218+wd%3AP169+wd%3AP112+wd%3AP1454+wd%3AP137+wd%3AP1029+wd%3AP225+wd%3AP274+wd%3AP1346+wd%3AP1120+wd%3AP498+wd%3AP856+wd%3AP1324+wd%3AP1581+%7D%0A++++%7D%0A++++OPTIONAL+%7B+%3Fitem+rdfs%3Alabel+%3Fname.+%7D%0A%7D%0A HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/13-6877fc25-3.js HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/48-e0f5f1c7-3.js HTTP/1.1" 200 None
DEBUG:searx.wikidata:request time 0:00:02.272877py
DEBUG:searx.engines:wikidata engine: Initialized
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/0-6516b28c-3.js HTTP/1.1" 200 None

generating indices...  genindex py-modindexdone
writing additional pages...  searchdone
copying images... [100%] dev/hello.dot
copying downloadable files... [100%] dev/reST.rst
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in dist/docs.
Copying tabs assets
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/2-46c8af5f-3.js HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/1-29bdd2fe-3.js HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/47-ae1d61aa-3.js HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://a-v2.sndcdn.com:443 "GET /assets/46-72510647-3.js HTTP/1.1" 200 None
DEBUG:searx.engines:soundcloud engine: Initialized
```
</details>

--- a/docs/dev/contribution_guide.rst
+++ b/docs/dev/contribution_guide.rst
@@ -141,6 +141,9 @@ Here is an example which makes a complete rebuild:
 live build
 ----------
 
+.. _sphinx-autobuild:
+   https://github.com/executablebooks/sphinx-autobuild/blob/master/README.md
+
 .. sidebar:: docs-clean
 
    It is recommended to assert a complete rebuild before deploying (use
@@ -156,8 +159,20 @@ changed.
    $ make docs-live
    ...
    The HTML pages are in dist/docs.
-   ... Serving on http://0.0.0.0:8080
+   ... Serving on http://0.0.0.0:8000
    ... Start watching changes
+
+Live builds are implemented by sphinx-autobuild_.  Use environment
+``$(SPHINXOPTS)`` to pass arguments to the sphinx-autobuild_ command.  Except
+option ``--host`` (which is always set to ``0.0.0.0``) you can pass any
+argument.  E.g to find and use a free port, use:
+
+.. code:: sh
+
+   $ SPHINXOPTS="--port 0" make docs-live
+   ...
+   ... Serving on http://0.0.0.0:50593
+   ...
 
 
 .. _deploy on github.io:

--- a/utils/makefile.sphinx
+++ b/utils/makefile.sphinx
@@ -79,7 +79,7 @@ quiet_cmd_sphinx = SPHINX    $@ --> file://$(abspath $(DOCS_DIST)/$5)
 	-b $2 -c $3 -d $(DOCS_BUILD)/.doctrees $4 $(DOCS_DIST)/$5
 
 quiet_cmd_sphinx_autobuild = SPHINX    $@ --> file://$(abspath $(DOCS_DIST)/$5)
-      cmd_sphinx_autobuild = PATH="$(PY_ENV_BIN):$(PATH)" $(PY_ENV_BIN)/sphinx-autobuild  $(SPHINX_VERBOSE) --open-browser --host 0.0.0.0 --port 8080 $(SPHINXOPTS)\
+      cmd_sphinx_autobuild = PATH="$(PY_ENV_BIN):$(PATH)" $(PY_ENV_BIN)/sphinx-autobuild  $(SPHINX_VERBOSE) --open-browser --host 0.0.0.0 $(SPHINXOPTS)\
 	-b $2 -c $3 -d $(DOCS_BUILD)/.doctrees $4 $(DOCS_DIST)/$5
 
 quiet_cmd_sphinx_clean = CLEAN     $@

